jerry-core/parser/regexp/re-compiler.c

```diff
@@ -388,7 +388,7 @@ re_parse_alternative (re_compiler_ctx_t *re_ctx_p, /**< RegExp compiler context
 
         ECMA_TRY_CATCH (empty,
                         re_parse_char_class (re_ctx_p->parser_ctx_p,
-                                             re_append_char_class,
+                                             (re_char_class_callback) re_append_char_class,
```

Comment:
Yes, thanks for the comments. I've updated the patch.
@LaszloLango : This modification will affect some helper functions, and more code refactoring needed. I think we could open a follow-up PR to do this kind of conversions.